packages/flutter/lib/src/widgets/modal_barrier.dart

```diff
@@ -228,12 +229,16 @@ class _AnyTapGestureRecognizer extends BaseTapGestureRecognizer {
     : super(debugOwner: debugOwner);
 
   VoidCallback? onAnyTapUp;
+  VoidCallback? onAnyTapCancel;
 
   @protected
   @override
   bool isPointerAllowed(PointerDownEvent event) {
-    if (onAnyTapUp == null)
+    if (onAnyTapUp == null) {
       return false;
+    } else if (onAnyTapCancel == null) {
```

Comment:
The current logic return false if one of onAnyTapUp and onAnyTapCancel is null, maybe they should be &&?